src/main/java/com/epam/aidial/core/service/ResourceService.java

```diff
@@ -442,24 +442,47 @@ public FileMetadata finishFileUpload(
     }
 
     public void computeResource(ResourceDescription descriptor, Function<String, String> fn) {
+        computeResource(descriptor, EtagHeader.ANY, fn);
+    }
+
+    public ResourceItemMetadata computeResource(ResourceDescription descriptor, EtagHeader etag, Function<String, String> fn) {
```

Comment:
I would add a Unit test. The logic has more branches need to be verified.

Reply:
All branches are already covered.

Reply:
Where are the branches covered? Could you please point out to a place?